eval/eval_api.go

```diff
@@ -22,7 +22,7 @@ type State struct {
 	macroState *object.Environment
 	env        *object.Environment
 	cache      Cache
-	extensions map[string]object.Extension
+	Extensions map[string]object.Extension
```

Comment:
I don't see why you are changing the visibility.

Reply:
because that's the most direct way to allow additions in the preinput hook, the test (or the other pr meanwhile) will make that more clear

Reply:
added test showing how it's used/why